software/base/src/test/java/org/apache/brooklyn/entity/machine/SetHostnameCustomizerTest.java

```diff
@@ -64,4 +68,20 @@ public void testGeneratedHostnameUsesPublicIpIfEmptyListOfPrivate() throws Excep
         
         assertEquals(customizer.generateHostname(machine), "ip-4-3-2-1-"+machine.getId());
     }
+    
+    @Test
+    public void testGeneratedHostnameIfNoIps() throws Exception {
+        SshMachineLocation machine = mgmt.getLocationManager().createLocation(LocationSpec.create(SshMachineLocationWithNoPublicIps.class)
+                .configure("address", "4.3.2.1"));
```

Comment:
It is compulsory configuration for an `SshMachineLocation` to have an address. I think that instantiation would have failed otherwise.